ui/src/shared/components/Notification.js

```diff
@@ -0,0 +1,93 @@
+import React, {Component, PropTypes} from 'react'
+import {connect} from 'react-redux'
+import {bindActionCreators} from 'redux'
+
+import classnames from 'classnames'
+
+import {dismissNotification as dismissNotificationAction} from 'shared/actions/notifications'
+
+class Notification extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      opacity: 1,
+      height: 0,
+      dismiss: false,
+    }
+  }
+
+  componentDidMount = () => {
+    const {notification: {duration}} = this.props
+
+    // Trigger animation in
+    const {height} = this.notificationRef.getBoundingClientRect()
+    this.setState({height})
+
+    if (duration >= 0) {
+      // Automatically dismiss notification after duration prop
+      this.dismissTimer = setTimeout(this.handleDismiss, duration)
```

Comment:
can we assign this to the class `state` and avoid setting state on the class instance itself?